src/test/java/seedu/address/logic/commands/CustomAssertionError.java

```diff
@@ -0,0 +1,44 @@
+package seedu.address.logic.commands;
+
+/**
+ * CustomAssertionError extends AssertionError to provide a more specific error indication
+ * for scenarios where a method is invoked inappropriately. It simplifies the process
+ * of throwing assertion errors with a default or custom message across the application.
+ */
+public class CustomAssertionError extends AssertionError {
+    // Static message for the error
+    private static final String DEFAULT_MESSAGE = "This method should not be called.";
```

Comment:
Perhaps a better name would be:
```suggestion
    private static final String INAPPROPRIATE_METHOD_CALL_MESSAGE = "This method should not be called.";
```